src/main/java/bio/terra/service/filedata/flight/ingest/IngestPopulateFileStateFromFileStep.java

```diff
@@ -1,116 +1,70 @@
 package bio.terra.service.filedata.flight.ingest;
 
-import bio.terra.common.FlightUtils;
 import bio.terra.model.BulkLoadFileModel;
-import bio.terra.model.BulkLoadRequestModel;
 import bio.terra.service.filedata.exception.BulkLoadControlFileException;
-import bio.terra.service.filedata.flight.FileMapKeys;
-import bio.terra.service.filedata.google.gcs.GcsBufferedReader;
-import bio.terra.service.filedata.google.gcs.GcsPdao;
-import bio.terra.service.job.JobMapKeys;
 import bio.terra.service.load.LoadService;
-import bio.terra.service.load.flight.LoadMapKeys;
-import bio.terra.service.resourcemanagement.google.GoogleBucketResource;
-import bio.terra.stairway.FlightContext;
-import bio.terra.stairway.FlightMap;
 import bio.terra.stairway.Step;
-import bio.terra.stairway.StepResult;
 import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
-import com.google.cloud.storage.Storage;
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
 
-// Populate the files to be loaded from the incoming array
-public class IngestPopulateFileStateFromFileStep implements Step {
+public abstract class IngestPopulateFileStateFromFileStep implements Step {
   private final LoadService loadService;
   private final int maxBadLines;
   private final int batchSize;
-  private final GcsPdao gcsPdao;
 
   public IngestPopulateFileStateFromFileStep(
-      LoadService loadService, int maxBadLines, int batchSize, GcsPdao gcsPdao) {
+      LoadService loadService, int maxBadLines, int batchSize) {
     this.loadService = loadService;
     this.maxBadLines = maxBadLines;
     this.batchSize = batchSize;
-    this.gcsPdao = gcsPdao;
   }
 
-  @Override
-  public StepResult doStep(FlightContext context) {
-    // Ensure that file ingestion works with extra key-value pairs
+  void readFile(BufferedReader reader, UUID loadId) throws IOException {
     ObjectMapper objectMapper =
         new ObjectMapper()
             .registerModule(new Jdk8Module())
             .registerModule(new JavaTimeModule())
             .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
```

Comment:
I know its not your change, but I think this should become a bean instantiated in `bio/terra/app/configuration/ApplicationConfiguration.java`. How it is now, we're creating a new `ObjectMapper` for each ingest instead of creating one at startup and sharing it across all ingest flights.

Reply:
Ok, cool -I think I've made this change.